File: Data Structures & Algorithms/sliding-window-maximum/submission-1.cpp
class Solution_brute_force_1 {
public:
    vector<int> maxSlidingWindow(vector<int>& nums, int k) {
        vector<int> ans;
        for(int i=0, j=k ; j<=nums.size() ; i++, j++) { // <= used since k j is exclusive end of window
            int maximum = INT_MIN;
            for(int x=i; x<j; x++) {
                maximum = max(maximum, nums[x]);
            }
            ans.push_back(maximum);
        }
        return ans;
    }
};

class Solution {
public:
    vector<int> maxSlidingWindow(vector<int>& nums, int k) {
        vector<int> ans;
        // k <= size of nums already given in condition
        for(int i=0, j=k-1 ; j<nums.size() ; i++, j++) { // j is inclusive index
            int maximum = INT_MIN;
            for(int x=i; x<=j; x++) {
                maximum = max(maximum, nums[x]);
            }
            ans.push_back(maximum);
        }
        return ans;
    }
};

File: Data Structures & Algorithms/subsets/submission-1.java
class Solution {
    private List<List<Integer>> ans = new ArrayList<>();
    public List<List<Integer>> subsets(int[] nums) {
        ans.add(new ArrayList<>());
        addSubsets(new ArrayList<>(), nums, 0);
        return ans;
    }
    public void addSubsets(ArrayList<Integer> already, int[] nums, int i) {
        for(;i<nums.length;i++) {
            List<Integer> temp = new ArrayList<>(already);
            temp.add(nums[i]);
            ans.add(temp);
            // ans.add((new ArrayList<>(already)).add(nums[i]));
            already.add(nums[i]);
            addSubsets(already, nums, i+1);
            already.remove(already.size()-1);
        }
    }
}
